src/demos.py

```diff
@@ -186,12 +186,13 @@ def pipeline_demo():
     limit = get_int_input("Set limit for data points to be processed (0=No limit)\n")
     limit = limit if limit > 0 else None
 
-    if limit is not None and DATABASE_TYPE == "S3":
-        choice = input(
-            "The output cannot be limited when uploading to S3.\nThe limit will be removed, and the pipeline will be executed on the full database.\n\nWould you like to continue? (y/n)\n"
-        )
-
-        if choice != "y" and choice != "Y":
+    if (
+        limit is not None
+        and get_database().DF_OUTPUT == "s3://amos--data--events/leads/enriched.csv"
+    ):
+        if get_yes_no_input(
```

Comment:
We should not be proceeding with the execution if answer is "no". 
```suggestion
        if get_yes_no_input(
            f"The output cannot be limited when uploading to {get_database().DF_OUTPUT}.\nThe limit will be removed, and the pipeline will be executed on the full database.\n\nWould you like to continue? (y/n)\n"
        ):
            limit = None
        else:
            return
```

Reply:
fixed